Modules/Core/Common/include/itkShapedNeighborhoodIterator.h

```diff
@@ -179,7 +179,7 @@ class ITK_TEMPLATE_EXPORT ShapedNeighborhoodIterator
   using typename Superclass::IndexValueType;
 
   /** An  iterator for the ShapedNeighborhood classes. */
-  struct Iterator : public ConstIterator
+  struct Iterator : public Superclass::ConstIterator
```

Comment:
Note: Instead of this adjustment (deriving from `Superclass::ConstIterator`), we could also restore `using ConstIterator = typename Superclass::ConstIterator;`